src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -118,7 +121,85 @@ class TextAnalyticsLanguageDetection(override val textAnalyticsOptions: Option[T
     }
 }
 
+object TextSentimentV4 extends ComplexParamsReadable[TextSentimentV4]
+class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsRequestOptions] = None,
+                      override val uid: String = randomUID("TextSentimentV4"))
+  extends TextAnalyticsSDKBase[SentimentScoredDocumentV4](textAnalyticsOptions)
+    with HasConfidenceScoreCol {
+  logClass()
+
+  override def outputSchema: StructType = SentimentResponseV4.schema
+
+  override protected val invokeTextAnalytics: String => TAResponseV4[SentimentScoredDocumentV4]= (text: String) =>{
+    val textSentimentResultCollection = textAnalyticsClient.analyzeSentimentBatch(
+      Seq(text).asJava, "en", textAnalyticsOptions.orNull)
+
+    def getConfidenceScore(score: SentimentConfidenceScores): SentimentConfidenceScoreV4 = {
+      SentimentConfidenceScoreV4(
+        score.getNegative,
+        score.getNeutral,
+        score.getPositive)
+    }
+
+    val textSentimentResult = textSentimentResultCollection.asScala.head
+    val documentSentiment = textSentimentResult.getDocumentSentiment();
+
+    val sentimentResult = if (textSentimentResult.isError){
+      None
+    } else {
+      Some(SentimentScoredDocumentV4(
+        documentSentiment.getSentiment().toString,
+        getConfidenceScore(documentSentiment.getConfidenceScores),
+        documentSentiment.getSentences.asScala.toList.map(sentenceSentiment =>
+          SentimentSentenceV4(
+            sentenceSentiment.getText,
+            sentenceSentiment.getSentiment.toString,
+            getConfidenceScore(sentenceSentiment.getConfidenceScores),
+            if(sentenceSentiment.getOpinions == null){
+              None
+            }else{
+              Some(sentenceSentiment.getOpinions.asScala.toList.map(op =>
+                OpinionV4(TargetV4(op.getTarget.getText,
+                  op.getTarget.getSentiment.toString,
+                  getConfidenceScore(op.getTarget.getConfidenceScores),
+                  op.getTarget.getOffset,
+                  op.getTarget.getLength)
+                  ,op.getAssessments.asScala.toList.map(assess =>
+                    AssessmentV4(assess.getText,
+                      assess.getSentiment.toString,
+                      getConfidenceScore(assess.getConfidenceScores),
+                      assess.isNegated,
+                      assess.getOffset,
+                      assess.getLength)))))
+            },
+            sentenceSentiment.getOffset,
+            sentenceSentiment.getLength
+          )),
+        documentSentiment.getWarnings.asScala.toList.map(warnings =>
+          WarningsV4(warnings.getMessage, warnings.getWarningCode.toString)

```

Comment:
This is getting too long:) Could we also define functions to transfer SentimentScoredDocumentV4/SentimentSentenceV4/etc?